### PR TITLE
vitess collectd draft

### PR DESCRIPTION
**Pull request by adkhare**
@bbeaudreault please can you review

### CODE REVIEW

File: vttablet_collectd.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+VITESS_CONFIG = {
+    'Host':           'localhost',
+    'Port':           15001,
+}
+
+def process_data(json_data):
+    epoch_time = util.get_epoch_time()
+
+    util.publish_metric(epoch_time, "vitess.appConnPoolAvailable", json_data['AppConnPoolAvailable']
+                        , 'counter', None)
```
Comment (bbeaudreault) [inline vttablet_collectd.py]: have not looked at all of the json data yet, but are these really all `counter` types? Typically counter means something that always increments for the lifetime of the process. I think many of these seem more like `gauge` types.

File: vtgate_collectd.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    util.create_metric(epoch_time, "vitess.vtgateInfoErrorCounts", json_data['VtgateInfoErrorCounts']
+                       , TAG_LIST_2)
+
+    util.create_metric(epoch_time, "vitess.vtgateInternalErrorCounts"
+                       , json_data['VtgateInternalErrorCounts'], TAG_LIST_2)
+
+    util.create_metric(epoch_time, "vitess.vttabletCallErrorCount", json_data['VttabletCallErrorCount']
+                       , TAG_LIST_3)
+
+    util.publish_metric(epoch_time, "vitess.vtgateApi.totalCount", json_data['VtgateApi']['TotalCount']
```
Comment (bbeaudreault) [inline vtgate_collectd.py]: this actually seems more like a counter time (based just on the name, i dont have access to the data right now)

Comment (adkhare) [inline vtgate_collectd.py]: Its still a gauge. counter gives a null value :-(

**Comment by harshit-gangal** [comment]
This looks good :)
Will this be packaged as plugin to collectd?

**Comment by adkhare** [comment]
Yes. It's now a collectd plugin.

**Comment by harshit-gangal** [comment]
Then try to get it published here https://collectd.org/wiki/index.php/Table_of_Plugins

**Pull request closed without merging**